### DISCUSSION TITLE
Node 代码执行 run failed: error: operation not permitted

### DISCUSSION THREAD

**Discussion by yuhp-zts** [langgenius/dify#3933] [General]
I created a workflow and want to use the output of the previous node as the input for code execution in this node. However, during debugging, the code execution node throws an error. The code in the code execution node is as follows：
def main(arg1):
    from datetime import datetime, timedelta
    import pytz,json
    text=json.loads(arg1)
    return {"text":text}

error：Node 代码执行 run failed: error: operation not permitted
the output of the previous node is ：{
  "arg1": "{\"start_date\": \"(datetime.now(pytz.timezone('Asia/Shanghai')) - timedelta(hours=5)).strftime('%Y-%m-%d %H:%M:%S')\", \"end_date\": \"(datetime.now(pytz.timezone('Asia/Shanghai'))).strftime('%Y-%m-%d %H:%M:%S')\"}"
}

**Comment by YidaHu** [comment]
@yuhp-zts You imported an uninstalled dependency package, I changed the code. It works.
<img width="408" alt="image" src="https://github.com/langgenius/dify/assets/13404367/dfe62127-7be9-4812-b7ab-7aaebd606729">


<img width="432" alt="image" src="https://github.com/langgenius/dify/assets/13404367/5ab5c0d4-b98c-43e2-8973-d6c72f082d49">

**Comment by crazywoola** [comment]
Please make sure following our language policy #1542

**Comment by lakako** [comment]
How to import third party dependencies? For example I want to import json5

**Reply by yuhp-zts** [reply]
Code execution tools have limited packages available. If you want to add a new package, I've heard that you need to modify the sandbox image, add the required package inside, and then repack it into a new image. However, I haven't actually done it myself. You can give it a try.

**Reply by shixiang1011** [reply]
> 如何导入第三方依赖项？例如，我想导入 json5

您好，请问导入第三方依赖项你有相关操作了吗
